server/models.py

```diff
@@ -113,21 +119,70 @@ class Participant(db.Model, TimestampMixin):
                           nullable=False)
     role = db.Column(db.Enum(*VALID_ROLES, name='role'), default=STUDENT_ROLE, nullable=False)
 
-    user = db.relationship("User", backref="participants")
+    user = db.relationship("User", backref="participations")
     course = db.relationship("Course", backref="participants")
+    notes = db.Column(db.String()) # For Section Info etc.
 
     def has_role(self, course, role):
         if self.course != course:
             return False
         return self.role == role
 
-    @hybrid_property
-    def is_course_staff(self):
-        return self.role in STAFF_ROLES
-
     def is_staff(self, course):
         return self.course == course and self.role in STAFF_ROLES
 
+    @staticmethod
+    def enroll_from_form(cid, form):
```

Comment:
Use `@transaction` to make sure `db.session.rollback()` is called if an exception is thrown